src/test/java/rx/internal/operators/OnSubscribeFromArrayTest.java

```diff
@@ -64,4 +65,15 @@ public void backpressure() {
         ts.assertCompleted();
     }
 
+    @Test
+    public void empty() {
+        Assert.assertSame(Observable.empty(), Observable.from(new Object[0]));
+    }
+
+    @Test
+    public void just() {
+        Observable<Integer> source = Observable.from(new Integer[] { 1 });
+        Assert.assertTrue(source.getClass().toString(), source instanceof ScalarSynchronousObservable);
```

Comment:
Right. That optimization is an implementation detail of performance, not of behavior. This means any time you change the optimization you have to change the unit test, despite the observed behavior of the API not changing one big.